### ISSUE TITLE
Remove discovery results when the bridge goes OFFLINE

### ISSUE THREAD

**Issue by afuechsel** [eclipse/smarthome#5204]
It would be a good extension to ESH that discovery results of a discovery service triggered by a bridge (e.g. the Hue bridge) would be automatically be removed from the inbox, when the bridge is going OFFLINE.  If they would remain in the inbox, they could be accepted and things being created that are not working as the bridge itself is OFFLINE. There might also be bindings, where the things could not be created without a successful communication with the bridge.

**Comment by sjka** [comment]
> There might also be bindings, where the things could not be created without a successful communication with the bridge.

That is true. However, then the question is why their discovery services don't remove their discovery results in that case themselves? There are several options, like a time-to-live for the result and/or a dedicated `removeOlderResults(long, ThingUID)` which would explicitly remove the results for that bridge. 

I'm not 100% sure yet if such an automatic removal would be suitable for all bindings (e.g. when they don't have any background discovery), but for sure easily doable in case of the Hue binding, as you mentioned it as an example.

I see though that it would be pointless to re-implement this kind of logic in lots of bindings again and again. That's just a source for errors and something a framework should prevent.

> There might also be bindings, where the things could not be created without a successful communication with the bridge.

That shouldn't be the case actually. The thing is just a database entry which gets created regardless of any handler configuration - and talking about configuration, I would expect the discovery service to supply the required configuration already in order to communicate with the device correctly. 
Therefore do you have concrete example what you meant by this?
